opt/api/utils/http/HttpStringBuilder.java

```diff
@@ -0,0 +1,57 @@
+package opt.api.utils.http;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+public class HttpStringBuilder {
+    
+    public static String build(BufferedReader in) throws IOException {
```

Comment:
Fixed it now:
```
        while ((line = in.readLine()) != null && !line.trim().isEmpty()) {
            requestBuilder.append(line).append("\r\n");
        }
        String PostHeaderCrlf = "\r\n";
        requestBuilder.append(PostHeaderCrlf);

        int contentLength = getBodyContentLength(requestBuilder.toString().trim());
        char[] buffer = new char[contentLength];
        int bytesRead = in.read(buffer, 0, contentLength);
```

Why was this so hard? It was exactly what I was saying - `readLine` waiting but the POST request sends a blank line with no return char 😢